tests/keithley_2700.py

```diff
@@ -0,0 +1,188 @@
+            self.ca.set_pv_value("KHLY2700_01:STAMP:SP", input)
+            self.ca.assert_that_pv_is("KHLY2700_01:STAMP", output)
+
+    @skipIf(IOCRegister.uses_rec_sim, "In rec sim this test fails")
+    def test_GIVEN_measurement_resolution_set_WHEN_read_THEN_measurement_resolution_is_as_expected(self):
+        self.ca.set_pv_value("KHLY2700_01:CHANNEL_END:SP", 110)
+        self.ca.set_pv_value("KHLY2700_01:DIGIT_RANGE:SP", 6)
+        self.ca.assert_that_pv_is("KHLY2700_01:DIGIT_RANGE", 6)
+        self.ca.assert_that_pv_is("KHLY2700_01:CHANNEL_END", 110)
+
+    @skipIf(IOCRegister.uses_rec_sim, "In rec sim this test fails")
+    def test_GIVEN_auto_range_set_WHEN_read_THEN_auto_range_is_as_expected(self):
+        self.ca.set_pv_value("KHLY2700_01:CHANNEL_END:SP", 110)
+        self.ca.set_pv_value("KHLY2700_01:AUTO_RANGE:SP", 0)
+        self.ca.assert_that_pv_is("KHLY2700_01:AUTO_RANGE", "ON")
+        self.ca.assert_that_pv_is("KHLY2700_01:CHANNEL_END", 110)
+
+    @skipIf(IOCRegister.uses_rec_sim, "In rec sim this test fails")
+    def test_GIVEN_measurement_mode_set_to_dc_current_WHEN_read_then_measurement_mode_is_dc_current(self):
+        expected_value = 26.0
+        self._lewis.backdoor_set_on_device("reading", expected_value)
+        expected_configuration = "DCI"
+        device_mode = self._lewis.backdoor_command(["device", "get_channel_param", "1", "mode"],
+                                                            returns_single_value=True)
+        #self.assertEquals(device_mode, expected_configuration)
+        self.ca.assert_that_pv_is("KHLY2700_01:MEAS:CURR:DC", expected_value)
+
+    @skipIf(IOCRegister.uses_rec_sim, "In rec sim this test fails")
+    def test_GIVEN_measurement_mode_set_to_dc_voltage_WHEN_read_then_measurement_mode_is_dc_voltge(self):
+        expected_value = 26.0
+        self._lewis.backdoor_set_on_device("reading", expected_value)
+        expected_configuration = "DCV"
+        self.ca.assert_that_pv_is("KHLY2700_01:MEAS:VOLT:DC", expected_value)
+        device_mode = self._lewis.backdoor_command(["device", "get_channel_param", "1", "mode"],
+                                                   returns_single_value=True)
+        #self.assertEquals(device_mode, expected_configuration)
+
+    @skipIf(IOCRegister.uses_rec_sim, "In rec sim this test fails")
+    def test_GIVEN_data_string_set_WHEN_read_THEN_data_string_is_as_expected(self):
+        self.ca.set_pv_value("KHLY2700_01:LIST_OF_ELEMENTS:SP", "READ,CHAN,TST")
+        self.ca.assert_that_pv_is("KHLY2700_01:LIST_OF_ELEMENTS:SP", "READ,CHAN,TST")
+        self._lewis.backdoor_set_on_device("reading", 45.0)
+        self._lewis.backdoor_set_on_device("channel_number", 110)
+        self._lewis.backdoor_set_on_device("time_stamp", 4)
+        self.ca.set_pv_value("KHLY2700_01:LIST_OF_ELEMENTS.PROC", 1)
+        self.ca.assert_that_pv_is("KHLY2700_01:LIST_OF_ELEMENTS", "45.0,110,4")
```

Comment:
The expected string is constructed from the input values. As previously, it's better to not hard code it. Extract the input values into variables and use those variables to construct the expected return string. It makes the test less fragile and gives a much better impression of how the input corresponds to the output.